scripts/o3de/o3de/register.py

```diff
@@ -398,12 +397,41 @@ def register_gem_path(json_data: dict,
                       gem_path: pathlib.Path,
                       remove: bool = False,
                       engine_path: pathlib.Path = None,
-                      project_path:  pathlib.Path = None) -> int:
+                      project_path:  pathlib.Path = None,
+                      force: bool = False) -> int:
```

Comment:
Should we add check options for the `register_gem_path` and `register_project_path` functions as well?

Reply:
a `check`/`dry-run` option would probably be good in the future for register in general - at least for anything that may fail due to compatibility.